project-management/Github.md

```diff
@@ -0,0 +1,77 @@
+# Managing public (or private) projects using GitHub
+
+We use a common set of labels across projects in GitHub. It is
+possible to precisely control this set of labels using
+the [settings GitHub app][github-app-settings] (needs to be enabled
+for your org). Use the [`settings.yml`](../.github/settings.yml) from
+this repository as a starting point.
+
+[github-app-settings]: https://github.com/probot/settings
+
+## Labels
+
+We use labels to classify tasks by priority, from P0 to P4:
+
+* **P0 (blocker)** Do not do anything else, this is a blocker.
+* **P1 (critical)** Must have.
+* **P2 (major)** Should have.
+* **P3 (minor)** Could have.
+* **P4 (trivial)** Someday/maybe. Consider closing as "wontfix".
+
+Labels are also useful to classify issues by type or by component. Use
+colours to distinguish whether a label designates a type, a component
+or something else.
+
+## Project leads
+
+Each project must have a project lead (maintainer). By convention, the
+PL is whoever appears at the top of [`CODEOWNERS`][github-codeowners]
+file, e.g.:
+
+```
+* @mboes
+```
+
+This has the effect that @mboes will automatically be tagged as
+a reviewer for all incoming PR's, unless later lines in the same file
+override that.
+
+[github-codeowners]: https://help.github.com/articles/about-codeowners/
+
+## Milestones
+
+We gate tickets with milestones. Milestones always have an associated
+due date.
+
+## Releases
+
+Every public release must include a `CHANGELOG.md` file, in the format
+of [keepachangelog]. New entries to the changelog are integral parts
+of bug fix and feature PR's.
+
+Each release has an associated tag of the form `v$VERSION`, GPG-signed
+by the project lead (use `git tag -s`).
+
+[keepachangelog]: https://keepachangelog.com/en/1.0.0/
+
+## Documentation
+
+Every document has a *document owner*. The owner is usually specified
+in the document itself. By default, the project lead is the owner for
+all documentation in the project.
+
+We do not use [wikis][github-wiki] for documentation. The reasons for
+this are well explained [elsewhere][yesod-documentation-thoughts].
+When using [GitHub pages][github-pages], we *do not* use a separate
+`gh-pages` branch to hold documentation. The reason is that this
```

Comment:
I've never figured out the workflow of Github pages with the `docs/` directory if it contains anything except purely static or jenkins-generated documentation (the main example for us is Haddock documentation, as well as Sphinx).

Is there any possibility at all? If so, we should probably have a short guide for it. If not, then we should propose an alternative workflow.

Reply:
Yes, I meant Jekyll :flushed: .

What I'm looking for is what to do when I'm not using Jekyll, which I'm not seeing in the links there. For Sphinx, I can simply use readthedocs. But, for Haddock?

In my personal project, what I've been doing is set up CI to push the compiled doc on the `gh-pages` branch. But we seem to be ruling this out. Maybe the answer is that we only publish Haddock documentation on Hackage? And avoid any customised documentation beyond what Jekyll can produce?

Reply:
> we only publish Haddock documentation on Hackage

Yes.

This is a project management page. It doesn't have recipes for all your documentation needs. We don't even always have a good solution for every use case at this point in time. Note that pushing *only generated* documentation to `gh-pages` is okay(ish), because that doesn't break the fundamental property that `master` and only `master` (not a set of disconnected branches) should be the ground truth. I'll add a sentence to reenforce that.